File: 2996-smallest-missing-integer-greater-than-sequential-prefix-sum/2996-smallest-missing-integer-greater-than-sequential-prefix-sum.cpp
class Solution {
public:
    int missingInteger(vector<int>& nums) {
        int seq_sum = nums[0];
        for(size_t i = 1; i < nums.size(); i++){
            if(nums[i] == nums[i - 1] + 1){
                seq_sum += nums[i];
            }else{
                break;
            }
        }
        std::unordered_set<int> numsSet(nums.begin() , nums.end());
        while(numsSet.count(seq_sum)){
            seq_sum++;
        }
        return seq_sum;
    }
};

File: 2996-smallest-missing-integer-greater-than-sequential-prefix-sum/2996-smallest-missing-integer-greater-than-sequential-prefix-sum.py
class Solution:
    def missingInteger(self, nums: List[int]) -> int:
        seq_sum = nums[0]
        for i in range(1 , len(nums)):
            if nums[i] == nums[i - 1] + 1:
                seq_sum += nums[i]
            else:
                break
        nums_set = set(nums)
        while seq_sum in nums_set:
            seq_sum += 1
        return seq_sum